pkg/providers/vsphere/vsphere.go

```diff
@@ -443,6 +450,80 @@ func (p *vsphereProvider) validateMachineConfigsNameUniqueness(ctx context.Conte
 	return nil
 }
 
+func (p *vsphereProvider) getCurrentMachineConfigDatastoreUsage(ctx context.Context, machineConfig *v1alpha1.VSphereMachineConfig, vsphereClusterSpec *Spec, cluster *types.Cluster) (diskGiB float64, err error) {
+	if cluster == nil {
+		return 0, nil
+	}
+	em, err := p.providerKubectlClient.SearchVsphereMachineConfig(ctx, machineConfig.Name, cluster.KubeconfigFile, vsphereClusterSpec.Cluster.GetNamespace())
+	if err != nil {
+		return 0, err
+	}
+	if len(em) > 0 {
+		return float64(em[0].Spec.DiskGiB), nil
+	}
+	return 0, nil
+}
+
+func (p *vsphereProvider) calculateDatastoreUsage(ctx context.Context, machineConfig *v1alpha1.VSphereMachineConfig, vsphereClusterSpec *Spec, cluster *types.Cluster, usage map[string]*datastoreUsage) error {
+	availableSpace, needGiB, err := p.getMachineConfigDatastoreRequirements(ctx, machineConfig, vsphereClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Count)
+	if err != nil {
+		return err
+	}
+	currentUsage, err := p.getCurrentMachineConfigDatastoreUsage(ctx, machineConfig, vsphereClusterSpec, cluster)
+	if err != nil {
+		return err
+	}
+	availableSpace += currentUsage
```

Comment:
why `availableSpace += currentUsage`?

Reply:
This helps to account for space that's already used by current machines. We calculate the NeedGiB based on the machine `diskGiB * count`, so if the machine config already exists (thus already taking up datastore space), we are double dipping during the calculation without this line.